clients/src/main/java/org/apache/kafka/clients/consumer/KafkaConsumer.java

```diff
@@ -753,6 +753,10 @@ public void subscribe(Pattern pattern, ConsumerRebalanceListener listener) {
     public void subscribe(Pattern pattern) {
         delegate.subscribe(pattern);
     }
+    @Override
+    public void subscribe(SubscriptionPattern pattern, ConsumerRebalanceListener callback) {}
```

Comment:
I expect we should be calling the delegate.subscribe(SubscriptionPattern..) here, otherwise the actual implementation in the AsyncKafkaConsumer won't be called. Similar to the subscribe(Pattern..) 
(This KafkaConsumer is the user-facing api, that ends up calling the Legacy or Async consumer via the delegate)